examples/nodejs/src/experimental-features/applicable-inputs.ts

```diff
@@ -0,0 +1,512 @@
+ */
+
+import {
+  MarloweState,
+  Party,
+  Contract,
+  Deposit,
+  Choice,
+  Input,
+  ChosenNum,
+  Environment,
+  Timeout,
+  getNextTimeout,
+  datetoTimeout,
+  Case,
+  Action,
+  Notify,
+  InputContent,
+  RoleName,
+  Token,
+  Bound,
+} from "@marlowe.io/language-core-v1";
+import {
+  applyAllInputs,
+  convertReduceWarning,
+  evalObservation,
+  evalValue,
+  inBounds,
+  Payment,
+  reduceContractUntilQuiescent,
+  TransactionWarning,
+} from "@marlowe.io/language-core-v1/semantics";
+import { AddressBech32, ContractId, PolicyId } from "@marlowe.io/runtime-core";
+import { RestClient } from "@marlowe.io/runtime-rest-client";
+import { WalletAPI } from "@marlowe.io/wallet";
+import * as Big from "@marlowe.io/adapter/bigint";
+import { Monoid } from "fp-ts/lib/Monoid.js";
+import * as R from "fp-ts/lib/Record.js";
+
+type ActionApplicant = Party | "anybody";
+
+export interface AppliedActionResult {
+  /**
+   * What inputs needs to be provided to apply the action
+   */
+  inputs: Input[];
+
+  /**
+   * What is the environment to apply the inputs
+   */
+  environment: Environment;
+  /**
+   * What is the new state after applying an action and reducing until quiescent
+   */
+  reducedState: MarloweState;
+  /**
+   * What is the new contract after applying an action and reducing until quiescent
+   */
+  reducedContract: Contract;
+  /**
+   * What warnings were produced while applying an action
+   */
+  warnings: TransactionWarning[];
+  /**
+   * What payments were produced while applying an action
+   */
+  payments: Payment[];
+}
+
+interface CanNotify {
+  type: "Notify";
+  policyId: PolicyId;
+  applyAction(): AppliedActionResult;
+}
+
+interface CanDeposit {
+  type: "Deposit";
+  policyId: PolicyId;
+  deposit: Deposit;
+
+  applyAction(): AppliedActionResult;
+}
+
+interface CanChoose {
+  type: "Choice";
+  policyId: PolicyId;
+  choice: Choice;
+
+  applyAction(choice: ChosenNum): AppliedActionResult;
+}
+
+/**
+ * This Applicable action is intended to be used when the contract is not in a quiescent state.
+ * This means that the contract is either timeouted, or it was just created and it doesn't starts with a `When`
+ */
+interface CanAdvance {
+  type: "Advance";
+  policyId: PolicyId;
+  applyAction(): AppliedActionResult;
+}
+
+export type ApplicableAction = CanNotify | CanDeposit | CanChoose | CanAdvance;
+
+function getApplicant(action: ApplicableAction): ActionApplicant {
+  switch (action.type) {
+    case "Notify":
+    case "Advance":
+      return "anybody";
+    case "Deposit":
+      return action.deposit.party;
+    case "Choice":
+      return action.choice.for_choice.choice_owner;
+  }
+}
+
+export async function getApplicableActions(
+  restClient: RestClient,
+  contractId: ContractId,
+  environment?: Environment
+): Promise<ApplicableAction[]> {
+  const contractDetails = await restClient.getContractById(contractId);
+  const currentContract = contractDetails.currentContract
```

Comment:
what does it means when it is closed ?  `currentContract == undefined` but we it to be the initial one ? why ?